src/cmd/src/options.rs

```diff
@@ -40,10 +54,187 @@ impl Options {
             Options::Cli(opts) => opts,
         }
     }
+
+    /// Load the configuration from multiple sources and merge them.
+    /// The precedence order is: environment variables > config file > default values.
```

Comment:
Not sure if we want to or is it possible to merge command line arguments as well here.

Also because our config file is specified via command line argument, do you think it has higher priority to environment variables. Imagine you call greptime with an explicit config file:

```sh
greptime start standalone -c myconfig.toml
```

And some of the config items didn't take effect because there is a conflict in envvars.

Reply:
Maybe the priority of the env should be higher than the command line. (env is highest).

Reply:
hmmm.. The reason why I assume command line arguments has higher priority is because they're written at last, say:

```
CONFIG=foo.toml greptime -c bar.toml
```

It makes me feel that `bar.toml` overrides `foo.toml` as `bar.toml` is specified after `foo.toml`. 

Perhaps we can check other application for how to deal with the priority.

Reply:
@sunng87, it seems that there is no convention of configuration order. The following popular projects have different behaviors:

- [etcd](https://etcd.io/docs/v3.4/op-guide/configuration/): config-file > cli > env;
- [kubelet](https://kubernetes.io/docs/tasks/administer-cluster/kubelet-config-file/): cli > config-file(for backwards compatibility);
- [viper](https://github.com/spf13/viper): the most famous configuration lib for Go, cli > env > config > default;
- [TiKV](https://tikv.org/docs/6.5/deploy/configure/tikv-command-line/): cli > config

Reply:
> It makes me feel that `bar.toml` overrides `foo.toml` as `bar.toml` is specified after `foo.toml`

I guess in this PR, we will read the configuration from `bar.toml`, but if users specify other envs like HTTP_ADDR, it will override the options inside `bar.toml`.

Reply:
> 

It looks you prefer: **config-file > cli > env > defaults**(to make it easy, there'is not default value for config file).

Reply:
I think it's:

cli > config file from cli > env > defaults

Reply:
@sunng87 got it, sounds reasonable.

Reply:
Modify in https://github.com/zyy17/greptimedb/commit/055d3672445c14f40e91cd134071b9bb7fba0fab#diff-207b5205d54878a73c2c6c9feb447957a4b4de179aca14fa2cc648d165a91835